impeller/renderer/render_pass.cc

```diff
@@ -142,9 +142,13 @@ bool RenderPass::SetVertexBuffer(BufferView vertex_buffers[],
     return false;
   }
 
-  pending_.vertex_buffer_count = vertex_buffer_count;
+  if (!vertex_buffers_start_.has_value()) {
+    vertex_buffers_start_ = vertex_buffers_.size();
+  }
+
+  pending_.vertex_buffers.length += vertex_buffer_count;
```

Comment:
you should absolutely not incrementally reserve a vector, that can degrade its re-allocation performance. Worst case scenario, O(n^2)

Reply:
Here's the case I was thinking.  Imagine it reallocates * 2.

```c++
// initial size 2, count = 1;
void foo(std::vector<int>& vec) {
  foo.push(1);
  // initial size 2, count = 2;
  foo.push(1);
  // reallocation, initial size 4, count = 3;
  foo.push(1);
  // initial size 4, count = 4;
  foo.push(1);
  // reallocation, initial size 8, count = 5;
}
```

A call to `foo.reserve(foo.capacity() + 4)` at the beginning of this function would avoid one reallocation, no?

Reply:
every draw has one vertex buffer though.

var buffers = []; // iniital size 0, initial capacity 0
for (var draw in draws) {
  for (var vb in draw.vbs) {
    buffers.reserve(size); // increase size by 1, re-allocate, copy.
    buffers.push_back(vb);
  }
}

This has ON^2 because each push_back requires us to fully reallocate the buffer.

Reply:
That's not quite what I was recommending.  I was recommending this:
```c++
  vertex_buffers_.reserve(vertex_buffers_.size() + vertex_buffer_count);
  for (size_t i = 0; i < vertex_buffer_count; i++) {
    vertex_buffers_.push_back(vertex_buffers[i]);
  }
```

We know how many push_backs we'll be giving, a reserve can avoid the situation where there is more than on realloc in the middle of the loop.

Reply:
> We know how many push_backs we'll be giving

we do not though. We only know the number of vertex buffers that have been recorded and that the current command has (which is always 1 for the 2d renderer).

Reply:
Okay, I'm convinced.  I think trying to outsmart the default reallocation algorithm here is counter productive since we can't holistically say what the final size should be.

Reply:
(I was thinking reserve would work with the doubling algorithm to be a noop, but that doesn't make sense and that's not how it works)